### PR TITLE
Specify "bios.hddOrder" during the CreateVMX step for the vmware-iso builder.

### PR DESCRIPTION
**Pull request by arizvisa**
Apparently ESX does some strange tampering with "bios.bootOrder" if "bios.hddOrder" is not defined resulting in "bios.bootOrder" being modified to just "cdrom" instead of what packer specifies it as ("hdd,cdrom"). This is strange because at the time the template is complete, the `CleanVMX` step should have disabled the cdrom so if ESX was trying to tamper with the boot order according to what devices were available, then "cdrom" should not be in its list.

Anyways, this hardcodes the "bios.hddOrder" field in the .vmx based on which disk device was specified using the suggestion by @GMZwinge which is expected to result in ESX leaving the "bios.bootOrder" as-is.

This also patches another issue that results in the cdrom-device not being properly removed by the `CleanVMX` step (common/step_clean_vmx.go). That issue will manifest itself if the cdrom-device is something other than IDE due to the way the regex was written.

This might close issue #6197.

### CODE REVIEW

**Comment by arizvisa** [comment]
kk. seems like the tests made it past the vmware-iso builder...

please wait user to confirm this on issue #6197 before merging.

File: builder/vmware/iso/step_create_vmx.go [hunk 1]
```diff
@@ -396,23 +399,28 @@ func (s *stepCreateVMX) Run(_ context.Context, state multistep.StateBag) multist
 
 	/// Use the disk adapter type that the user specified to tweak the .vmx
 	//  Also sync the cdrom adapter type according to what's common for that disk type.
+	//  XXX: If the cdrom type is modified, make sure to update common/step_clean_vmx.go
```
Comment (mwhooker) [inline builder/vmware/iso/step_create_vmx.go]: I don't think a comment is going to have the enforcement we're looking for. any way to turn this into a test?

Comment (arizvisa) [reply builder/vmware/iso/step_create_vmx.go]: Afaict, turning it into a test will still need some kind of enforcement as if a new device type is added (not likely, but still possible) both the test itself and the steps too will still need knowledge of what types are available in order to guarantee detaching the iso.

This also becomes awkward as a user can also affect the type out-of-band using the `vmx_data` option which can bypass these checks entirely. This is why I assume the author chose to use a regex to select and detach all iso types (that packer knows about) which should cover all the cases where the step_clean_vmx.go step might not detach the iso.

This would be necessary unless someone refactors the device type selection to expose the available types (for disk and cdrom) as some sort of data instead of using if-tests/string-comparisons and regexes like the rest of the builders do (parallels comes to mind as doing this too).

In summary, the coupling between steps that attach and detach the iso as a result of packer needing knowledge of what device types a user might choose will still require someone to synchronize the the type checks in more than one place whether it's in a test or a step.

Comment (SwampDragons) [reply builder/vmware/iso/step_create_vmx.go]: I'm in favor of refactoring this selection to use some kind of data structure and actually fixing the problem, rather than continuing to throw band-aids onto it.
